mpas_analysis/ocean/climatology_map_waves.py

```diff
@@ -0,0 +1,526 @@
+            climatology['peakWavePeriod'] = \
+                1.0/climatology['timeMonthly_avg_peakWaveFrequency']
+
+        if 'timeMonthly_avg_significantWaveHeight' in climatology:
+            climatology['significantWaveHeight'] = \
+                climatology['timeMonthly_avg_significantWaveHeight']
+
+        return climatology
+
+        # }}}
+    # }}}
+
+
+class RemapEra5ObservedWaveClimatology(RemapObservedClimatologySubtask):  # {{{
+    """
+    A subtask for reading and remapping ERA5 wave observations
+    """
+    # Authors
+    # -------
+    # Steven Brus, Xylar Asay-Davis
+
+    def __init__(self, parentTask, seasons, fileName, outFilePrefix,
+                 variableList, subtaskName,
+                 comparisonGridNames=['latlon']):
+        # {{{
+        '''
+        Construct one analysis subtask for each plot (i.e. each season and
+        comparison grid) and a subtask for computing climatologies.
+
+        Parameters
+        ----------
+        parentTask :  ``AnalysisTask``
+            The parent (master) task for this subtask
+
+        seasons : list of str
+           A list of seasons (keys in ``constants.monthDictionary``) over
+           which the climatology should be computed.
+
+        fileName : str
+            The name of the observation file
+
+        outFilePrefix : str
+            The prefix in front of output files and mapping files, typically
+            the name of the field being remapped
+
+        variableList : list
+            List of observational variables to remap
+
+        comparisonGridNames : list of {'latlon', 'antarctic', 'arctic'},
+            optional
+            The name(s) of the comparison grid to use for remapping.
+
+        subtaskName : str, optional
+            The name of the subtask
+        '''
+        # Authors
+        # -------
+        # Steven Brus
+
+        self.variableList = variableList
+
+        # call the constructor from the base class (AnalysisTask)
+        super().__init__(parentTask, seasons, fileName, outFilePrefix,
+                         comparisonGridNames=['latlon'],
+                         subtaskName=subtaskName)
+        # }}}
+
+    def get_observation_descriptor(self, fileName):  # {{{
+        '''
+        get a MeshDescriptor for the observation grid
+
+        Parameters
+        ----------
+        fileName : str
+            observation file name describing the source grid
+
+        Returns
+        -------
+        obsDescriptor : ``MeshDescriptor``
+            The descriptor for the observation grid
+        '''
+        # Authors
+        # -------
+        # Steven Brus
+
+        # create a descriptor of the observation grid using the lat/lon
+        # coordinates
+        obsDescriptor = LatLonGridDescriptor.read(fileName=fileName,
+                                                  latVarName='latitude',
+                                                  lonVarName='longitude')
```

Comment:
To handle the periodic fix, this will need to become:
```suggestion
        dsObs = self.build_observational_dataset(fileName)
        obsDescriptor = LatLonGridDescriptor.read(ds=dsObs,
                                                  latVarName='latitude',
                                                  lonVarName='longitude')
```